src/Hal/Resource.cs

```diff
@@ -115,7 +115,8 @@ public Resource(object? state)
         /// <returns>The string representation of the current instance.</returns>
         public string ToString(JsonSerializerSettings jsonSerializerSettings)
         {
-            jsonSerializerSettings.Converters = converters;
+            if (jsonSerializerSettings.Converters == null || jsonSerializerSettings.Converters.Count == 0)
```

Comment:
you need this _jsonSerializerSettings.Converters == null_ ? in JsonSerializerSettings's constructor i think they always initialize it

Reply:
It seems you're correct, removed it. Thanks!